pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -174,6 +174,17 @@ func (a *ClusterDeploymentValidatingAdmissionHook) validateCreate(admissionSpec
 
 	// TODO: Put Create Validation Here (or in openAPIV3Schema validation section of crd)
 
+	if !validateIngressList(&newObject.Spec) {
+		message := "If defining the Ingress list, it must include a default ingress entry"
```

Comment:
If I need to include a "default" ingress entry, would the `IngressDomain` be the same as the `BaseDomain`? Or once ingress entries are provided the client would have to ensure that one of them is named "default", regardless of its domain?

We currently limit the number of router shards on the clusters-service to 2, but the labels are arbitrarily entered by the user. Wondering if I need to hard-code one of them as "default" or whether I should silently provide the "default" one using the `BaseDomain`.

/cc @jhernand

Reply:
Whoops, didn't mean to trigger the bot

/uncc @jhernand

Reply:
@vkareh looking at the default ingress for a 4.0 cluster, the ingressDomain is set to null in the spec, and then the operator sets one in the status.

Reply:
what i had in mind was that you could alter the clusteringress object named 'default', so if you wanted it changed from apps.example.com you could do that. although, the specific case i had in mind was just marking the default ingress as "internal only" (ie backed by a non-internet-facing ELB).

Reply:
if you're creating additional ingress objects, it sort-of makes sense that the default is the BaseDomain marked as internal.

What if I need to create separate internal shards? We limit to 2 currently, but we don't specify whether they need to be internal/external: they can both be internal if they so choose. How do I send the router shard scheme (internal vs internet-facing, for each ingress object) to Hive?

Reply:
i'm trying to catch up with the latest changes that have gone into this are of openshift, so forgive my vagueness.

setting internal vs external is dependent on the ingress object (now apparently renamed) exposing that tunable. the object as of this writing seems to only allow setting the domain https://github.com/openshift/api/blob/master/config/v1/types_ingress.go#L24 .

i'm installing a new cluster to investigate these changes and how they'll affect the desired functionality as exposed via hive

@vkareh where are you getting this limit of '2' from? this is the first i hear of it.

Reply:
@joelddiaz - no, we impose the limit ourselves in our cluster service, so nothing upstream - it doesn't relate to Hive (we just call Hive to provision the clusters)